src/app/pages/wallets/wallet-details/pages-wallet-details.component.ts

```diff
@@ -0,0 +1,129 @@
+import { ActivatedRoute } from '@angular/router';
+import { Observable, Subscription } from 'rxjs';
+import { TDataState } from 'src/app/common/http/common.http.types';
+import { SystemNotificationsService } from 'src/app/common/utils/system-notifications/system-notifications.service';
+import { WalletTransactionType } from 'src/app/domains/wallets/domains.wallets.types';
+import { PagesWalletsManagementService } from '../management/pages-wallets-management.service';
+import { WalletsManagementItem } from '../management/pages-wallets-wallets-management-item.model';
+import { PagesWalletsManagementEditorComponent } from '../management/wallet-editor/pages-wallets-management-editor.component';
+import { IWalletModalData } from '../management/wallet-editor/pages-wallets-management-editor.types';
+import { WalletsDetailsTransaction } from './pages-wallet-details-item.model';
+import { PagesWalletDetailsService } from './pages-wallet-details.service';
+
+@Component({
+  selector: 'app-pages-wallet-details',
+  templateUrl: './pages-wallet-details.component.html',
+  styleUrls: ['./pages-wallet-details.component.scss'],
+})
+export class PagesWalletDetailsComponent implements OnInit, OnDestroy {
+
+  public selectTransactionsTypes: Record<string, WalletTransactionType | ''> = {
+    'All transactions': '',
+    'Incomes': WalletTransactionType.Incomes,
+    'Expenses': WalletTransactionType.Expenses,
+  }
+
+  public selectedTransactionType: Record<string, WalletTransactionType | ''> = this.selectTransactionsTypes;
+
+  public transactionsTypeForm = new FormControl<WalletTransactionType | ''>('');
+
+  private transactionTypeSub?: Subscription;
+
+  public walletTransactionType: typeof WalletTransactionType = WalletTransactionType;
+
+  public displayedColumns: string[] = ['date', 'description', 'amount', 'actions'];
+
+  public walletsManagementItem?: WalletsManagementItem;
+
+  public displayedTransactions: WalletsDetailsTransaction[] = [];
+
+  public walletsDetailsData: TDataState<WalletsDetailsTransaction[]> = {
+    data: null,
+    hasError: false,
+    isLoading: true,
+  }
+
+  public constructor(
+    private readonly pagesWalletDetailsService: PagesWalletDetailsService,
+    private readonly myWalletsService: PagesWalletsManagementService,
+    private readonly systemNotificationsService: SystemNotificationsService,
+    private readonly dialog: MatDialog,
+    private readonly activatedRoute: ActivatedRoute,
+  ) { }
+
+  public ngOnInit(): void {
+    this.activatedRoute.data.subscribe( ({ wallet }) => {
+      if (!(wallet instanceof WalletsManagementItem)) {
+        return;
+      }
+
+      this.walletsManagementItem = wallet;
+      this.initWalletDetails(wallet);
+    });
+
+    this.transactionTypeSub = this.transactionsTypeForm.valueChanges.subscribe((data: WalletTransactionType | '' | null) => {
+      this.filterTransactions(data ?? '');
+    });
+  }
+
+  private initWalletDetails(wallet: WalletsManagementItem): void {
```

Comment:
It is more like `getWalletTransactions` I think. And only needed argument is `wallet.id`